js/src/common/Application.tsx

```diff
@@ -330,15 +380,15 @@ export default class Application {
       let responseText;
 
       if (original) {
-        responseText = original(xhr.responseText);
+        responseText = original(xhr, options);
```

Comment:
This seems like a change in behavior

Reply:
Yes, it was. The behaviour didn't match with the official Mithril typings, as we expected to pass the response text rather than the XHR object.

I'm changing the union type to a custom interface, and changing this back to fix it.